modules/programs/htop.nix

```diff
@@ -368,49 +537,60 @@ in {
     vimMode = mkOption {
       type = types.bool;
       default = false;
-      description = "Vim key bindings.";
+      description = ''
+        Deprecated. Please use programs.htop.settings.vim_mode instead.
+
+        Vim key bindings.
+      '';
     };
   };
 
   config = mkIf cfg.enable {
+    warnings = optional (cfg.settings == { }) ''
```

Comment:
This looks like a dubious check, maybe in the default settings' prefix cfg.X with `lib.warn "warning" cfg.X` isntead. Also maybe add a release note. I dont know when the deprecation will be removed. Maybe just after the release of nixos.

Reply:
Agreed, this one is ugly.

A few days ago I already pushed an added release note. Now also a commit to redo the configuration a bit. Moved defaults into the `settings` attrset and show a deprecation warning on any (deprecated) option that is explicitly specified in user config. Messages are much more clear now and should properly inform what to do, without relying on reading releases notes too much :)

If this PR is good to go, I'll draft up another one to remove all deprecated things (to merge after 21.05 release, or when suitable).